applications/CompressiblePotentialFlowApplication/custom_elements/transonic_perturbation_potential_flow_element.cpp

```diff
@@ -541,6 +589,48 @@ void TransonicPerturbationPotentialFlowElement<TDim, TNumNodes>::CalculateLeftHa
 
 }
 
+template <int TDim, int TNumNodes>
+void TransonicPerturbationPotentialFlowElement<TDim, TNumNodes>::CalculateLeftHandSideExtendedElement(
+    MatrixType& rLeftHandSideMatrix,
+    const ProcessInfo& rCurrentProcessInfo)
+{
+    if (rLeftHandSideMatrix.size1() != TNumNodes + 1 || rLeftHandSideMatrix.size2() != TNumNodes + 1)
+    {
+        rLeftHandSideMatrix.resize(TNumNodes + 1, TNumNodes + 1, false);
+    }
+    rLeftHandSideMatrix.clear();
+
+    ElementalData<TNumNodes, TDim> data;
+
+    // Calculate shape functions
+    GeometryUtils::CalculateGeometryData(GetGeometry(), data.DN_DX, data.N, data.vol);
+
+    const double density = ComputeDensity(rCurrentProcessInfo);
+    const double DrhoDu2 = ComputeDensityDerivative(density, rCurrentProcessInfo);
+
+    // Computing local velocity
+    const array_1d<double, 3> free_stream_velocity = rCurrentProcessInfo[FREE_STREAM_VELOCITY];
+    array_1d<double, TDim> velocity = PotentialFlowUtilities::ComputeVelocity<TDim,TNumNodes>(*this);
+    for (unsigned int i = 0; i < TDim; i++)
+    {
+        velocity[i] += free_stream_velocity[i];
+    }
+
+    const BoundedVector<double, TNumNodes> DNV = prod(data.DN_DX, velocity);
+
+    BoundedMatrix<double, TNumNodes, TNumNodes> current_lhs = data.vol * density * prod(data.DN_DX, trans(data.DN_DX));
+    current_lhs += data.vol * 2 * DrhoDu2 * outer_prod(DNV, trans(DNV));
+
+    for (int i = 0; i < TNumNodes; i++)
+    {
+        for (int j = 0; j < TNumNodes; j++)
+        {
+            rLeftHandSideMatrix(i, j) = current_lhs(i, j);
+        }
+    }
```

Comment:
I like this idea to fill in the matrix! You can do exactly this in CalculateLeftHandSideNormalElement to make it work for both cases (inlet and non inlet) and avoid implementing an extra function (for now).